plugins/modules/cloud/smartos/vmadm.py

```diff
@@ -604,7 +659,7 @@ def main():
             'zfs_root_compression', 'zpool'
         ],
         'bool': [
-            'archive_on_delete', 'autoboot', 'debug', 'delegate_dataset',
```

Comment:
This change breaks backwards compatibility. The option needs to be deprecated first.

Could you please move that part to another PR? This is the only thing that cannot be backported to stable-1 (which from now on only accepts bugfixes and no longer new deprecations).